lib/tools/adb-commands.js

```diff
@@ -14,4 +17,86 @@ methods.initZipAlign = async function() {
   this.binaries.zipalign = await this.getSdkBinaryPath("zipalign");
 };
 
+methods.getApiLevel = async function () {
+  log.info("Getting device API level");
+  try {
+    return this.shell(['getprop', 'ro.build.version.sdk']);
+  } catch (e) {
+    log.error(`Error getting device API level. Original error: ${e.message}`);
+    throw new Error(`Error getting device API level. Original error: ${e.message}`);
+  }
+};
+
+methods.isDeviceConnected = async function () {
+  let devices = await this.getConnectedDevices();
+  return devices.length > 0;
+};
+
+methods.mkdir = async function (remotePath) {
+  return this.shell(['mkdir', '-p', remotePath]);
+};
+
+methods.isValidClass = function (classString) {
+  // some.package/some.package.Activity
+  return new RegExp(/^[a-zA-Z0-9\./_]+$/).exec(classString);
+};
+
+methods.forceStop = async function (pkg) {
+  return this.shell(['am', 'force-stop', pkg]);
+};
+
+methods.clear = function (pkg) {
+  return this.shell(['pm', 'clear', pkg]);
+};
+
+methods.stopAndClear = async function (pkg) {
+  try {
+    await this.forceStop(pkg);
+    await this.clear(pkg);
+  } catch (e) {
+    log.error(`Cannot stop and clear ${pkg}. Original error: ${e.message}`);
```

Comment:
we might want to add a function `logAndThrow(msg)` to appium-support since this is a common pattern. then you could just do:

```js
logAndThrow(`Cannot stop and clear ${pkg}. Original error: ${e.message}`);
```